Justfile

```
@@ -77,8 +77,8 @@ install:
 # Remove temporary files
 [group('lifecycle')]
 clean:
-    rm -rf .venv .pytest_cache .mypy_cache .ruff_cache .coverage htmlcov
-    find . -type d -name "__pycache__" -exec rm -r {} +
+    uv run python -c "import shutil; shutil.rmtree('.venv', ignore_errors=True)"
```

Using `uv run` creates a .venv if none existed. Repeated execution of `just clean` would create it just to delete it. The following might offer a slight improvement.
```suggestion
    uvx python -c "import shutil; shutil.rmtree('.venv', ignore_errors=True)"
```

---

Just curious @bittner, wouldn't `rm -rf .venv` suffice instead of using `shutil` (in addition to PyClean) to stay platform-independent? AFAIK *just* runs in `sh` on all platforms by default.

---

According to [the just documentation](https://just.systems/man/en/prerequisites.html) you need Cygwin, Git for Windows, or similar on Windows. If instead you want to rely on bare Windows you can configure PowerShell or cmd as the shell. This will leave you without a `rm` command.

---

I see, thanks!

I am not aware of a way to run *just* in powershell without adjusting the justfile further (adding `set windows-shell`) or running *just* each time with additional command line arguments. Which leaves this solution a bit unsatisfactory. Or am I missing another option?

---

If you look at the full spectrum, is the Python solution more portable or one with Unix-style commands? Which one will work regardless of an adjusted or untouched justfile? – Feels like a weird political discussion. :face_with_diagonal_mouth:

---

Sorry, don't get me wrong. I am not criticizing your improvements, I was just checking that I am not missing anything. I haven't worked much with *just* yet. It surely is more pythonic that way for sure!